src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java

```diff
@@ -214,13 +217,129 @@ void setupKernel(ExtensionContext context) throws Exception {
+        NoOpPathOwnershipHandler.register(kernel);
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel,
+                EventFleetStatusServiceTest.class.getResource("onlyMain.yaml"));
+        kernel.getContext().put(MqttClient.class, mqttClient);
+        kernel.getContext().put(ThingGroupHelper.class, thingGroupHelper);
+
+        // Mock out cloud communication
+        GreengrassServiceClientFactory mgscf = mock(GreengrassServiceClientFactory.class);
+        GreengrassV2DataClient mcf = mock(GreengrassV2DataClient.class);
+        lenient().when(mcf.resolveComponentCandidates(any(ResolveComponentCandidatesRequest.class)))
+                .thenThrow(ResourceNotFoundException.class);
+        lenient().when(mgscf.fetchGreengrassV2DataClient()).thenReturn(mcf);
+        kernel.getContext().put(GreengrassServiceClientFactory.class, mgscf);
+
+        componentNamesToCheck.clear();
+        kernel.getContext().addGlobalStateChangeListener((service, oldState, newState) -> {
+            if (service.getName().equals(FleetStatusService.FLEET_STATUS_SERVICE_TOPICS)
+                    && newState.equals(State.RUNNING)) {
+                fssRunning.countDown();
+            }
+            if (service.getName().equals(DeploymentService.DEPLOYMENT_SERVICE_TOPICS)
+                    && newState.equals(State.RUNNING)) {
+                deploymentServiceRunning.countDown();
+                deploymentService = (DeploymentService) service;
+                IotJobsHelper iotJobsHelper = deploymentService.getContext().get(IotJobsHelper.class);
+                iotJobsHelper.setIotJobsClientWrapper(mockIotJobsClientWrapper);
+            }
+            if (service.getName().equals("main")
+                    && newState.equals(State.FINISHED)) {
+                mainFinished.countDown();
+            }
+            componentNamesToCheck.add(service.getName());
+        });
+        // set required instances from context
+        deviceConfiguration =
+                new DeviceConfiguration(kernel, "ThingName", "xxxxxx-ats.iot.us-east-1.amazonaws.com", "xxxxxx.credentials.iot.us-east-1.amazonaws.com", "privKeyFilePath",
+                        "certFilePath", "caFilePath", "us-east-1", "roleAliasName");
+        kernel.getContext().put(DeviceConfiguration.class, deviceConfiguration);
+        // pre-load contents to package store
+        Path localStoreContentPath =
+                Paths.get(EventFleetStatusServiceTest.class.getResource("local_store_content").toURI());
+        PreloadComponentStoreHelper.preloadRecipesFromTestResourceDir(localStoreContentPath.resolve("recipes"), kernel.getNucleusPaths().recipePath());
+        copyFolderRecursively(localStoreContentPath.resolve("artifacts"), kernel.getNucleusPaths().artifactPath(), REPLACE_EXISTING);
+        kernel.launch();
+
+        Thread.sleep(25000); // make sure Nucleus finished launching then verify\
+        // assertTrue(fssRunning.await(10, TimeUnit.SECONDS));
```

Comment:
clean this up? you can just wait for count down latches to finish instead of sleeping